### PR TITLE
Implement the Top Artists use case.

### PR DESCRIPTION
**Pull request by jackwiebe0**
The presenter has several errors because the ViewManagerModel has not yet been implemented (TODO in lab 11-19). The getTopArtists function in the data access object has been started, but I need more clarification before I can finish it. I have also added Javadoc descriptions to many of the functions in this use case.

### CODE REVIEW

File: src/data_access/UserDataAccessObject.java
Comment (jackwiebe0) [inline src/data_access/UserDataAccessObject.java:1]: I've left TODOs to remind myself where this function still needs work

File: src/use_case/TopArtist/TopArtistDataAccessInterface.java
Comment (jackwiebe0) [inline src/use_case/TopArtist/TopArtistDataAccessInterface.java:1]: I renamed TopArtist to TopArtists to reduce singular/plural confusion. I also changed the package name to top_artists to reflect the Week 5 clean architecture package structure.

**Review by ashleychristen — COMMENTED** [review]
This looks wonderful! thank you for adding todo and javadoc descriptions so I could fully understand what you were doing. We can fix those errors tomorrow. I think all thats left to do is the view!

**Review by coderveda27 — APPROVED** [review]
thanks for finishing up your use case! It is looking good, follows java style conventions, clean architecture and SOLID principles as well. We will thoroughly test this later on after the TODOs are completed

**Pull request merged**